src/write.c

```diff
@@ -800,49 +762,50 @@ static avifImage * avifImageCopyAndPad(const avifImage * srcImage, uint32_t dstW
-                // them, mimic the source image's state by removing our copy.
-                avifFree(dstImage->yuvPlanes[yuvPlane]);
-                dstImage->yuvPlanes[yuvPlane] = NULL;
-                dstImage->yuvRowBytes[yuvPlane] = 0;
-                continue;
-            }
-            avifCopyAndPadPlane(dstImage->yuvPlanes[yuvPlane],
-                                dstImage->yuvRowBytes[yuvPlane],
-                                (yuvPlane == AVIF_CHAN_Y) ? dstImage->width : dstUvWidth,
-                                (yuvPlane == AVIF_CHAN_Y) ? dstImage->height : dstUvHeight,
-                                srcImage->yuvPlanes[yuvPlane],
-                                srcImage->yuvRowBytes[yuvPlane],
-                                (yuvPlane == AVIF_CHAN_Y) ? srcImage->width : srcUvWidth,
-                                (yuvPlane == AVIF_CHAN_Y) ? srcImage->height : srcUvHeight,
-                                avifImageUsesU16(srcImage) ? 2 : 1);
-        }
     }
-
     if (srcImage->alphaPlane) {
         const avifResult allocationResult = avifImageAllocatePlanes(dstImage, AVIF_PLANES_A);
         if (allocationResult != AVIF_RESULT_OK) {
             avifImageDestroy(dstImage);
             return NULL;
         }
-        avifCopyAndPadPlane(dstImage->alphaPlane,
-                            dstImage->alphaRowBytes,
-                            dstImage->width,
-                            dstImage->height,
-                            srcImage->alphaPlane,
-                            srcImage->alphaRowBytes,
-                            srcImage->width,
-                            srcImage->height,
-                            avifImageUsesU16(srcImage) ? 2 : 1);
+    }
+    for (int plane = AVIF_CHAN_Y; plane <= AVIF_CHAN_A; ++plane) {
+        const uint8_t * srcRow = avifImagePlane(srcImage, plane);
+        const uint32_t srcRowBytes = avifImagePlaneRowBytes(srcImage, plane);
+        const uint32_t srcPlaneWidth = avifImagePlaneWidth(srcImage, plane);
+        const uint32_t srcPlaneHeight = avifImagePlaneHeight(srcImage, plane); // 0 for A if no alpha and 0 for UV if 4:0:0.
+        const size_t srcPlaneWidthBytes = (size_t)srcPlaneWidth << (srcImage->depth > 8);
+
+        uint8_t * dstRow = avifImagePlane(dstImage, plane);
+        const uint32_t dstRowBytes = avifImagePlaneRowBytes(dstImage, plane);
+        const uint32_t dstPlaneWidth = avifImagePlaneWidth(dstImage, plane);
+        const uint32_t dstPlaneHeight = avifImagePlaneHeight(dstImage, plane); // 0 for A if no alpha and 0 for UV if 4:0:0.
+        const size_t dstPlaneWidthBytes = (size_t)dstPlaneWidth << (dstImage->depth > 8);
+
+        for (uint32_t j = 0; j < srcPlaneHeight; ++j) {
+            memcpy(dstRow, srcRow, srcPlaneWidthBytes);
+
+            // Pad columns.
+            if (dstPlaneWidth > srcPlaneWidth) {
+                if (avifImageUsesU16(srcImage)) {
```

Comment:
Nit: Here we use `avifImageUsesU16(srcImage)`, but we use `(srcImage->depth > 8)` in line 778 and `(dstImage->depth > 8)` in line 784. These three expressions have the same value. It would be good to use the same expression in all three places, or save the value in a local variable and use the variable in all three places.

We should at least use the same expression in line 778 and here.